src/Microsoft.Health.SqlServer/DefaultSqlConnectionBuilder.cs

```diff
@@ -58,17 +58,20 @@ public DefaultSqlConnectionBuilder(IOptions<SqlServerDataStoreConfiguration> opt
     public string DefaultDatabase { get; }
 
     /// <inheritdoc />
+    [SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Caller must dispose result.")]
     public SqlConnection GetSqlConnection(string initialCatalog = null, int? maxPoolSize = null)
     {
         SqlConnectionStringBuilder builder = GetConnectionStringBuilder(initialCatalog, maxPoolSize);
         return new SqlConnection(builder.ToString()) { RetryLogicProvider = _retryProvider };
     }
 
     /// <inheritdoc />
-    [Obsolete($"Use {nameof(GetSqlConnection)} instead.")]
     [SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Caller must dispose result.")]
-    public Task<SqlConnection> GetSqlConnectionAsync(string initialCatalog = null, int? maxPoolSize = null, CancellationToken cancellationToken = default)
-        => Task.FromResult(GetSqlConnection(initialCatalog, maxPoolSize));
+    public async Task<SqlConnection> GetSqlConnectionAsync(string initialCatalog = null, int? maxPoolSize = null, CancellationToken cancellationToken = default)
```

Comment:
This pattern makes sense to me. It's like the `Stream` APIs with `Read` and `ReadAsync` methods.